pkg/input/arm.go

```diff
@@ -148,13 +135,34 @@ func (l *armConfiguration) Errors() []error {
 	for _, resource := range l.discovered {
 		errs = append(errs, resource.errors()...)
 	}
-	return errs
+	return append(errs, l.expressionEvalErrs...)
 }
 
 func (l *armConfiguration) Type() *Type {
 	return Arm
 }
 
+func (l *armConfiguration) process() {
+	// Set of all existing resources for the resolver.
+	resourceSet := map[string]struct{}{}
+	for id := range l.discovered {
+		resourceSet[id] = struct{}{}
+	}
+	refResolver := arm_ReferenceResolver{
```

Comment:
This is just a code move, it was named this before. But I think you're probably right, we've changed its purpose a bit. I guess it was named ReferenceResolver before because of its resourceId special case support, to ensure that cross-resource references were appropriately "linked" by being normalized to the same string representation?

Now that it's really doing something more general, evaluating ARM expressions, I'll try to name it something oriented around that. Holler if you disagree!